src/analyses/variable-sensitivity/array_abstract_object.cpp

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************\
+
+ Module: analyses variable-sensitivity
+
+ Author: Thomas Kiley, thomas.kiley@diffblue.com
+
+\*******************************************************************/
+
+#include <util/std_types.h>
+#include <util/std_expr.h>
+#include <analyses/variable-sensitivity/abstract_enviroment.h>
+#include "array_abstract_object.h"
+
+
+/*******************************************************************\
+
+Function: array_abstract_objectt::array_abstract_objectt
+
+  Inputs:
+   type - the type the abstract_object is representing
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+array_abstract_objectt::array_abstract_objectt(const typet &t):
+  abstract_objectt(t)
+{
+  assert(t.id()==ID_array);
```

Comment:
To keep the constructor consistent so can use in the abstract object factory: https://github.com/diffblue/cbmc/pull/708/files#diff-376bc05dbdc3369030faf77361786f1fR100